torch/_functorch/_aot_autograd/functional_utils.py

```diff
@@ -134,7 +134,14 @@ def are_all_mutations_under_no_grad_or_inference_mode(t):
 # Note: "storage mutations" coming from set_() are a type of metadata mutation. So:
 # - check_only_storage_mutation=True: only return true if there was a storage mutation
 # - check_only_storage_mutation=Flse: return true if there was any metadata mutation (including a storage mutation)
-def has_metadata_mutation(f_arg, arg, *, check_only_storage_mutation: bool):
+def has_metadata_mutation(
```

Comment:
Hmm - "was the tensor non-contiguous" is technically unrelated to whether or not there was a metadata mutation, so it's a bit confusing to put the logic here (in our case, we already ban metadata mutations on subclass inputs - we just want to additionally ban data-mutations on subclass inputs, when we detect that the subclass was (originally) non-contiguous.

Reply:
Would a more targeted check like this work?
```
mutates_data = has_data_mutation(f_arg) # from existing code
if mutates_data and not arg.is_contiguous() and is_traceable_wrapper_subclass(arg):
    raise RuntimeError(...)
```

Reply:
Yeah I definitely agree it's confusing. I was thinking we could count the stride mutation that results from functionalization as a "metadata mutation", but that's certainly an abuse of that terminology :p

> Would a more targeted check like this work?

It'll work in general, but I will likely have to revisit this for NJT since we do want to mutate "non-contiguous" inputs for KV-caching. I expect them to have contiguous strides but be non-contiguous in the "has holes" sense, so this check will be over-eager in failing those.